Webapp/sources/services/all_compounds.py

```diff
@@ -149,3 +150,76 @@ def mol_weight_from_smiles(smiles: str) -> float:
     """
     # MolWt accounts for the average across isotopes but ExactMolWt only takes the most abundant isotope.
     return round(Descriptors.MolWt(Chem.MolFromSmiles(smiles)), 2)
+
+
+def from_cas(
+    cas: str, workbook: models.WorkBook = None
+) -> Union[models.Compound, models.NovelCompound]:
+    """
+    Returns a compound or novel compound from a cas string
+
+    Args:
+        cas - cas number of a compound
+        workbook - if looking for a novel compound, the workbook we look in
+
+    Returns:
+        models.Compound or models.NovelCompound: The compound object retrieved from the database.
+                                                 Returns None if no matching compound is found.
+    """
+    compound = services.compound.from_cas(cas)
+    if not compound and workbook:
```

Comment:
Is this meant to mean `if compound is None and workbook is Nonw`?

Reply:
No, this is meant to be if there is no compound but there is a workbook. Would it be clearer written as: `if workbook is not None and compound is None`?

Reply:
I see. That might be a bit more explicit. Or perhaps `if workbook and not compound`? I don't know why, just in my head the current version reads logically like `if not (compound and workbook)` if that makes sense

Reply:
Yeah, that makes sense. I'll update both to what you've suggested.